src/main/java/com/redhat/devtools/intellij/common/utils/ConfigWatcher.java

```diff
@@ -38,30 +38,22 @@ public class ConfigWatcher implements Runnable {
     private WatchService service;
 
     public interface Listener {
-        void onUpdate(ConfigWatcher source, Config config);
+        void onUpdate(Config updatedConfig);
     }
 
-    public ConfigWatcher(String config, Listener listener) {
-        this(Paths.get(config), listener);
+    public ConfigWatcher(Listener listener) {
+        this(Config.getKubeconfigFilenames().stream().map(Paths::get).toList(), listener, new HighSensitivityRegistrar());
     }
 
-    public ConfigWatcher(Path config, Listener listener) {
-        this(List.of(config), listener);
-    }
-
-    public ConfigWatcher(List<Path> configs, Listener listener) {
-        this(configs, listener, new HighSensitivityRegistrar());
-    }
-
-    public ConfigWatcher(List<Path> configs, Listener listener, HighSensitivityRegistrar registrar) {
+    protected ConfigWatcher(List<Path> configs, Listener listener, HighSensitivityRegistrar registrar) {
         this.configs = configs;
         this.listener = listener;
         this.registrar = registrar;
     }
 
     @Override
     public void run() {
-        watch((Config config) -> listener.onUpdate(this, config));
+        watch(listener::onUpdate);
```

Comment:
notify listeners of a change in the client config (not individual kubeconf files)